[Add]ボウリングスコア計算プログラムの作成

jun-kondo opened the pull request:
## 概要
- ボウリングスコア計算プログラムを作成しました。下記チェック項目がパスしていることは確認しています。
*1/23追記 メソッド部分を別ファイルに切り出して、若干コードを修正しました。プログラムの実行結果とRuboCopの結果を改めて確認しています。

レビューよろしくお願いいたします🙇‍♂️
## チェック項目
- [x] 「プログラム実行の例」にある入力例は全て実行結果を載せること。
- [x]  rubocop-fjordのチェックが全てパス

## プログラム実行結果
<details><summary>一回目</summary>

```
# jun.k @ mcbk in ~/workspace/fjord/ruby-practices/03.bowling on git:my-bowling x [10:53:01]
$ ./bowling.rb 6,3,9,0,0,3,8,2,7,3,X,9,1,8,0,X,6,4,5
139

# jun.k @ mcbk in ~/workspace/fjord/ruby-practices/03.bowling on git:my-bowling x [10:53:27]
$ ./bowling.rb 6,3,9,0,0,3,8,2,7,3,X,9,1,8,0,X,X,X,X
164

# jun.k @ mcbk in ~/workspace/fjord/ruby-practices/03.bowling on git:my-bowling x [10:53:34]
$ ./bowling.rb 0,10,1,5,0,0,0,0,X,X,X,5,1,8,1,0,4
107

# jun.k @ mcbk in ~/workspace/fjord/ruby-practices/03.bowling on git:my-bowling x [10:53:41]
$ ./bowling.rb 6,3,9,0,0,3,8,2,7,3,X,9,1,8,0,X,X,0,0
134

# jun.k @ mcbk in ~/workspace/fjord/ruby-practices/03.bowling on git:my-bowling x [10:53:48]
$ ./bowling.rb 6,3,9,0,0,3,8,2,7,3,X,9,1,8,0,X,X,1,8
144

# jun.k @ mcbk in ~/workspace/fjord/ruby-practices/03.bowling on git:my-bowling x [10:53:54]
$ ./bowling.rb X,X,X,X,X,X,X,X,X,X,X,X
300

# jun.k @ mcbk in ~/workspace/fjord/ruby-practices/03.bowling on git:my-bowling x [10:54:02]
$ ./bowling.rb X,X,X,X,X,X,X,X,X,X,X,2
292

# jun.k @ mcbk in ~/workspace/fjord/ruby-practices/03.bowling on git:my-bowling x [10:54:17]
$ ./bowling.rb X,0,0,X,0,0,X,0,0,X,0,0,X,0,0
50
```
## Rubocop実行結果
```
# jun.k @ mcbk in ~/workspace/fjord/ruby-practices/03.bowling on git:my-bowling x [11:12:43]
$ rubocop bowling.rb
Inspecting 1 file
.

1 file inspected, no offenses detected
```

</details>
1/23 修正後の実行実行結果

```
# jun.k @ mcbk in ~/workspace/fjord/ruby-practices/03.bowling on git:my-bowling x [13:16:30] 
$ ./bowling.rb 6,3,9,0,0,3,8,2,7,3,X,9,1,8,0,X,6,4,5
139

# jun.k @ mcbk in ~/workspace/fjord/ruby-practices/03.bowling on git:my-bowling x [13:17:28] 
$ ./bowling.rb 6,3,9,0,0,3,8,2,7,3,X,9,1,8,0,X,X,X,X
164

# jun.k @ mcbk in ~/workspace/fjord/ruby-practices/03.bowling on git:my-bowling x [13:17:35] 
$ ./bowling.rb 0,10,1,5,0,0,0,0,X,X,X,5,1,8,1,0,4
107

# jun.k @ mcbk in ~/workspace/fjord/ruby-practices/03.bowling on git:my-bowling x [13:17:43] 
$ ./bowling.rb 6,3,9,0,0,3,8,2,7,3,X,9,1,8,0,X,X,0,0
134

# jun.k @ mcbk in ~/workspace/fjord/ruby-practices/03.bowling on git:my-bowling x [13:17:49] 
$ ./bowling.rb 6,3,9,0,0,3,8,2,7,3,X,9,1,8,0,X,X,1,8
144

# jun.k @ mcbk in ~/workspace/fjord/ruby-practices/03.bowling on git:my-bowling x [13:17:56] 
$ ./bowling.rb X,X,X,X,X,X,X,X,X,X,X,X
300

# jun.k @ mcbk in ~/workspace/fjord/ruby-practices/03.bowling on git:my-bowling x [13:18:03] 
$ ./bowling.rb X,X,X,X,X,X,X,X,X,X,X,2
292

# jun.k @ mcbk in ~/workspace/fjord/ruby-practices/03.bowling on git:my-bowling x [13:18:09] 
$ ./bowling.rb X,0,0,X,0,0,X,0,0,X,0,0,X,0,0
50

```
## Rubocop実行結果

```
# jun.k @ mcbk in ~/workspace/fjord/ruby-practices/03.bowling on git:my-bowling x [13:12:42] 
$ rubocop bowling.rb                                
Inspecting 1 file
.

1 file inspected, no offenses detected

# jun.k @ mcbk in ~/workspace/fjord/ruby-practices/03.bowling on git:my-bowling x [13:16:28] 
$ rubocop bowling_methods.rb
Inspecting 1 file
.

1 file inspected, no offenses detected
```

jun-kondo committed:
[Add]ボウリングスコア計算プログラムの作成

ボウリングスコア計算プログラムを作成しました。

jun-kondo committed:
[Modify]不要なコメントを削除

jun-kondo committed:
[Modify]メソッド部分をbowling_methods.rbに切り出し

メソッドのテストがやりやすいようにメソッド部分を別ファイルに切り出しました。

jun-kondo committed:
[Add]スコア計算メソッド用のファイルを作成

jun-kondo committed:
[Modify]strikeメソッドをリファクタリング

summertree128 commented on 03.bowling/bowling_methods.rb:
```diff
@@ -0,0 +1,24 @@
+      if frame[0] == 10
+        strike_bonus(frames, index) + 10
+      elsif frame.sum == 10
+        frames[index + 1][0] + 10
```
なんとなくですが、`strike_bonus`があるのであればスペアの方もメソッドになっていると読みやすいかなと思いました。修正必須ではありません。

jun-kondo replied:
冗長かと思い削ってしまいましたが、あったほうが良かったですね。
修正しました!!
41e6ca6f94f79c410768a8c3f91de21b99252f21

summertree128 commented on 03.bowling/bowling_methods.rb:
```diff
@@ -0,0 +1,24 @@
+  end
+end
+
+def culc_score(frames)
```
`calculation` の略であれば`calc`でしょうか。

jun-kondo replied:
修正しました!!スペルミス気をつけます。。。
41e6ca6f94f79c410768a8c3f91de21b99252f21

summertree128 commented on 03.bowling/bowling_methods.rb:
```diff
@@ -0,0 +1,24 @@
+end
+
+def culc_score(frames)
+  point = 0
```
scoreなのかpointなのか、語彙をしっかり定義したい感はありますね。動作に問題はないと思いますが。
今修正して欲しいわけではないので、今後この辺をより気にするようにしてみてください。

jun-kondo replied:
確かにその通りですね。スコアに統一しておきます!!
41e6ca6f94f79c410768a8c3f91de21b99252f21

summertree128 commented on 03.bowling/bowling.rb:
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require_relative 'bowling_methods'
```
現時点だとファイルを分割するメリットがないので、各メソッドも`bowling.rb`に定義した方が良いとは思います。
修正するかどうかはお任せします。

jun-kondo replied:
ありがとうございます!!メソッド含めて１ファイルに収めました!!
41e6ca6f94f79c410768a8c3f91de21b99252f21

summertree128 reviewed (approved):
レビューしました。簡潔に書けていて良い実装だと思います。
いくつかコメントしてあるので、余力があれば修正してみてもらっても良いですし、コメントの内容さえ理解してもらえばそのままでも大丈夫です。
提出物としてはOKです〜。

jun-kondo committed:
[Modified]指摘箇所を修正

1. ポイントの変数名をscoreに統一
2. スペアメソッドを追加
3. メソッドを切り出したファイルを削除
4. スペルミスを修正 (x: culc => o: calc)

jun-kondo commented:
レビューありがとうございました!!
OKとのことでしたが、マージする前にご指摘箇所修正するように致しました。
任意での対応とのことでしたのでこのままマージさせて頂きます!!
